pkgs/compiler-lib/compiler/commands/test.rkt

```diff
@@ -517,11 +518,10 @@
           v))]))
 
 (define (normalize-module-path p)
-  (cond
-   [(path? p) (path->string p)]
-   [(and (pair? p) (eq? 'submod (car p)))
-    (list* 'submod (normalize-module-path (cadr p)) (cddr p))]
-   [else p]))
+  (match p
+    [(? path?) `(file ,(path->string p))]
```

Comment:
Do we really want to convert this to a string?

Reply:
`normalize-module-path` currently serves two purposes:

1. Printing the path as a string for logging or display purposes.
2. Processing the path to use it as an argument in subprocess calls.

Using `serialize` instead of `normalize-module-path` might be more appropriate for handling `rt-module` and `test-module`. If so, is it necessary to recover `normalize-module-path`?

Reply:
The problem with (2) from the comments I linked is that paths don't round-trip through strings in the general case (I forget what change I made for that though: it's probably in the PR comments I linked somewhere).